src/win.c

```diff
@@ -1128,6 +1131,9 @@ void win_on_factor_change(session_t *ps, struct managed_win *w) {
 	// Focus needs to be updated first, as other rules might depend on the focused
 	// state of the window
 	win_update_focused(ps, w);
+	w->blur_foreground =
```

Comment:
Need to add damage when blur_foreground changes

Reply:
I *think* 4bccb48 (based on `win_{determine,set}_blur_background`) should solve this